tensorbay/client/job.py

```diff
@@ -0,0 +1,126 @@
+        finished_at: The time when the Job is finished.
+        status: The status of the Job.
+        error_message: The error message of the Job.
+        result: The result of the Job.
+        description: The description of the Job.
+
+    """
+
+    _T = TypeVar("_T", bound="Job")
+
+    _repr_type = ReprType.INSTANCE
+    _repr_maxlevel = 2
+
+    _repr_attrs: Tuple[str, ...] = (
+        "title",
+        "description",
+        "arguments",
+        "created_at",
+        "started_at",
+        "finished_at",
+        "status",
+        "error_message",
+    )
+
+    title: str = attr()
+    job_id: str = attr(key=camel)
+    job_type: str = attr(key=camel)
+    arguments: Dict[str, Any] = attr()
+    created_at: int = attr(key=camel)
+    started_at: int = attr(key=camel)
+    finished_at: Optional[int] = attr(key=camel)
+    status: str = attr()
+    error_message: str = attr(key=camel)
+    _result: Optional[Dict[str, Any]] = attr(key="result")
+    description: Optional[str] = attr(default="")
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        update_info: Callable[[str], Dict[str, Any]],
+        title: str,
+        job_id: str,
+        job_type: str,
+        arguments: Dict[str, Any],
+        created_at: int,
+        started_at: int,
+        finished_at: Optional[int],
+        status: str,
+        error_message: str,
+        result: Optional[Dict[str, Any]],
+        description: Optional[str] = "",
+    ) -> None:
+        self.title = title
+        self.update_info = update_info
+        self.job_id = job_id
+        self.job_type = job_type
+        self.arguments = arguments
+        self.created_at = created_at
+        self.started_at = started_at
+        self.finished_at = finished_at
+        self.status = status
+        self.error_message = error_message
+        self._result = result
+        self.description = description
+
+    def _repr_head(self) -> str:
+        return f'{self.__class__.__name__}("{self.job_id}")'
+
+    @classmethod
+    def loads(cls: Type[_T], contents: Dict[str, Any]) -> _T:
+        """Loads a :class:`Job` instance for the given contents.
+
+        Arguments:
+            contents: A dict containing the information of the Job::
+
+                {
+                    "title": <str>
+                    "jobId": <str>
+                    "jobType": <str>
+                    "arguments": <object>
+                    "createdAt": <int>
+                    "startedAt": <int>
+                    "finishedAt": <int>
+                    "status": <str>
+                    "errorMessage": <str>
+                    "result": <object>
+                    "description": <str>
+                }
+
+        Returns:
+            A :class:`Job` instance containing all the information in the given contents.
+
+        """
+        return common_loads(cls, contents)
+
+    def update(self, update_until_complete: bool) -> None:
```

Comment:
I think `update_until_complete` is too long. Also the function name is `update`, the keyword starts with `update`, it is redundant.
As a reference:
In python multithread and multiprocess system, it use the keyword `blocking` to indicate that the function will be blocked until the lock released. But the word `blocking` is also not clear for this function, we need to add detailed doc and example for understanding.

Reply:
In my opinion, `until_complete` is enough.